[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52865 [quant][graphmode][fx][fp16] Add fp16 support for silu
* #52864 [quant][graphmode][fx][fp16] Add fp16 support for tanh
* #52863 [quant][graphmode][fx][fp16] Add fp16 support for sigmoid
* #52862 [quant][graphmode][fx][fp16] Add fp16 support for layer_norm
* #52811 [quant][graphmode][fx][fp16] Add fp16 support for torch.sum
* #52810 [quant][graphmode][fx][fp16] Add fp16 support for div
* #52809 [quant][graphmode][fx][fp16] Add fp16 support for sub
* #52808 [quant][graphmode][fx] Add support for fp16 bmm pattern
* **#52807 [quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests**
* #52714 [quant][graphmode][fx][fp16] Add fp16 support for {add|mul}{_relu}

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

jerryzh168 committed:
[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2129,31 +2137,37 @@ def forward(self, x, y):
 
 
     @skipIfNoFBGEMM
     def test_add(self):
```
maybe `_test_binary_op_int8_impl` and `_test_binary_op_float16_impl` could become their own tests, and same for the others?

jerryzh168 replied:
hmmm, it's a trade off I think, we can split but then we'll have more tests and it also increases complexity. but i don't have a very strong preference, please let me know if you feel it's worthwhile to make this trade off.

jerryzh168 replied:
will land for now, we can split in a separate PR if needed.

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][test][refactor] Refactoring binary op tests to split int8 and float16 tests"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655617](https://our.internmc.facebook.com/intern/diff/D26655617)

[ghstack-poisoned]
